lib/matplotlib/cbook/__init__.py

```diff
@@ -365,6 +389,13 @@ def process(self, s, *args, **kwargs):
                     proxy(*args, **kwargs)
                 except ReferenceError:
                     self._remove_proxy(proxy)
+                # this does not capture KeyboardInterrupt, SystemExit,
+                # and GeneratorExit
+                except Exception as e:
```

Comment:
I'm ok to leave this as is for now (we should just drop Py2 and use WeakMethods anyways :-))